hcl2template/parser.go

```diff
@@ -277,6 +292,47 @@ func (cfg *PackerConfig) Initialize() hcl.Diagnostics {
 	return diags
 }
 
+func (cfg *PackerConfig) evaluateDatasources(validationOnly bool) hcl.Diagnostics {
+	res := map[string]cty.Value{}
+	var diags hcl.Diagnostics
+
+	for ref, _ := range cfg.DataSources {
+		datasource, startDiags := cfg.startDatasource(cfg.dataStoreSchemas, ref)
+		diags = append(diags, startDiags...)
+		if diags.HasErrors() {
+			continue
+		}
+
+		value := cty.Value{}
+		if validationOnly {
+			placeholderValue := getSpecValue(datasource.OutputSpec()[ref.Type])
```

Comment:
I think that we should do a different datasource implementation per datasource type here: 
```suggestion
			placeholderValue := getSpecValue(datasource.OutputSpec())
```

Reply:
This is because `datasource.OutputSpec()` returns the ObjectSpec which is a map and, as I implemented thinking of a single value, the map will always have 1 element to the type, as I did with the amazon-ami example. 
But making the changes you suggested with multiple values in a struct, this will have to change anyway.

Reply:
Ah gotcha !